### ISSUE TITLE
[Update Request]: StefansTools.grepWin

### ISSUE THREAD

**Issue by RadAd** [microsoft/winget-pkgs#113741]
### What type of update are you requesting?

A new version of an existing package

### Current Package Identifier

StefansTools.grepWin

### Package Version

2.0.1232

### Please describe the changes you would like to see

My previous request was not updated correctly. https://github.com/microsoft/winget-pkgs/issues/113395

This package has historically used the build number in the version instead of the published version number.

The update should have been 2.0.1232, not 2.0.15.

**Comment by mhartkorn** [comment]
The version winget thinks is the newest version is critically broken with handling read-only files. winGrep will just skip and not search them unless "include symbolic links" is checked which is *not the default*.

The versioning scheme in winget is plain wrong. The official version scheme is 2.0.15.1232 for the current version from June 2023 and e.g. 2.0.*8.*1054 for older versions.

**Comment by Dragon1573** [comment]
- Duplicate of https://github.com/microsoft/winget-pkgs/issues/113395
- Resolved by https://github.com/microsoft/winget-pkgs/pull/113440

Issue can be closed.

**Comment by RadAd** [comment]
#113440 doesn't resolve this issue, it caused this issue.
The issue is winget has changed its versioning scheme for grepWin and has messed up future updates.
It would have been better to continue the old scheme.

**Comment by Dragon1573** [comment]
There are 2 ways to solve this:

1. Remove all previous versions of this package.
2. Change the version number to continue the further versioning.

These 2 methods can be done directly on github.dev , maybe you can achieve this yourself! :smile:

**Comment by DigitalJustin** [comment]
When this was fixed, I feel like version 2.0.15 was missed.  There is no DisplayVersion in [https://github.com/microsoft/winget-pkgs/blob/master/manifests/s/StefansTools/grepWin/2.0.15/StefansTools.grepWin.installer.yaml](url) so winget shows as follows:

![image](https://github.com/microsoft/winget-pkgs/assets/6685546/ec514911-c04d-4c7c-8509-9fa69deb4c7c)

Based on the registry
![image](https://github.com/microsoft/winget-pkgs/assets/6685546/a9d7e93f-63f2-429a-8730-29609b8b593f)

It would suggest there should be 
```
AppsAndFeaturesEntries:
- DisplayVersion: 2.0.1232
```

**Comment by RadAd** [comment]
I have the same issue:
```
> winget list --id StefansTools.grepWin
Name        Id                   Version  Available Source
----------------------------------------------------------
grepWin x64 StefansTools.grepWin > 2.0.12 2.0.15    winget
```
```
> reg query HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{75A64F1A-4A9F-44C9-957E-4036384C434B}

HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{75A64F1A-4A9F-44C9-957E-4036384C434B}
    AuthorizedCDFPrefix    REG_SZ    
    Comments    REG_SZ    
    Contact    REG_SZ    
    DisplayVersion    REG_SZ    2.0.1232
    HelpLink    REG_SZ    
    HelpTelephone    REG_SZ    
    InstallDate    REG_SZ    20230925
    InstallLocation    REG_SZ    C:\Users\xxx\AppData\Local\Programs\grepWin\
    InstallSource    REG_SZ    D:\
    ModifyPath    REG_EXPAND_SZ    MsiExec.exe /I{75A64F1A-4A9F-44C9-957E-4036384C434B}
    Publisher    REG_SZ    Stefans Tools
    Readme    REG_SZ    
    Size    REG_SZ    
    EstimatedSize    REG_DWORD    0x7fc
    UninstallString    REG_EXPAND_SZ    MsiExec.exe /I{75A64F1A-4A9F-44C9-957E-4036384C434B}
    URLInfoAbout    REG_SZ    
    URLUpdateInfo    REG_SZ    
    VersionMajor    REG_DWORD    0x2
    VersionMinor    REG_DWORD    0x0
    WindowsInstaller    REG_DWORD    0x1
    Version    REG_DWORD    0x20004d0
    Language    REG_DWORD    0x409
    DisplayName    REG_SZ    grepWin x64

```

The changes made to use the other versioning scheme have really messed this up.
